### ISSUE TITLE
Error: Failed to load config "prettier" to extend from. 

### ISSUE THREAD

**Issue by yubaoquan** [iamturns/eslint-config-airbnb-typescript#138]
```
[Error - 下午12:51:12] ESLint stack trace:
[Error - 下午12:51:12] Error: Failed to load config "prettier" to extend from.
Referenced from: D:\rs-test\minigrep\node_modules\eslint-config-airbnb-typescript\.eslintrc.js
    at configInvalidError (D:\rs-test\minigrep\node_modules\eslint\lib\cli-engine\config-array-factory.js:295:9)
    at ConfigArrayFactory._loadExtendedShareableConfig (D:\rs-test\minigrep\node_modules\eslint\lib\cli-engine\config-array-factory.js:876:23)
    at ConfigArrayFactory._loadExtends (D:\rs-test\minigrep\node_modules\eslint\lib\cli-engine\config-array-factory.js:776:25)
    at ConfigArrayFactory._normalizeObjectConfigDataBody (D:\rs-test\minigrep\node_modules\eslint\lib\cli-engine\config-array-factory.js:715:25)
    at _normalizeObjectConfigDataBody.next (<anonymous>)
    at ConfigArrayFactory._normalizeObjectConfigData (D:\rs-test\minigrep\node_modules\eslint\lib\cli-engine\config-array-factory.js:660:20)
    at _normalizeObjectConfigData.next (<anonymous>)
    at ConfigArrayFactory.loadInDirectory (D:\rs-test\minigrep\node_modules\eslint\lib\cli-engine\config-array-factory.js:508:28)
    at CascadingConfigArrayFactory._loadConfigInAncestors (D:\rs-test\minigrep\node_modules\eslint\lib\cli-engine\cascading-config-array-factory.js:367:46)
    at CascadingConfigArrayFactory.getConfigArrayForFile (D:\rs-test\minigrep\node_modules\eslint\lib\cli-engine\cascading-config-array-factory.js:288:18)

```

version: 
    "eslint-config-airbnb-typescript": "10.0.0",

Should you add it to dependencies instead of devDependencies?

**Comment by iamturns** [comment]
There are no `prettier` settings within the published `eslint-config-airbnb-typescript`. This looks to be a problem with your projects customised eslint config. Do you have prettier settings, but without the config added as a dependency?

**Comment by yubaoquan** [comment]
My project doesn't use prettier. There's no files related to prettier at all. But I have to add prettier to dependencies to overcome this issue. Because `.eslintrc.js` in eslint-config-airbnb-typescript needs prettier at run time.

```
ESLint: Failed to load config "prettier" to extend from. Referenced from: xxxxxx\node_modules\eslint-config-airbnb-typescript\.eslintrc.js. Please see the 'ESLint' output channel for details.
```

**Comment by iamturns** [comment]
This is unusual. I can see that the `eslint-config-airbnb-typescript/.eslintrc.js` file is being referenced here. This file should only be used internally by `eslint-config-airbnb-typescript` to lint itself.

It should be referencing either the `index.js` or `base.js` file.

Can you please share your ESLint config?

**Comment by yubaoquan** [comment]
```json

{
  "parser": "@typescript-eslint/parser",
  "extends": ["airbnb-typescript/base"],
  "plugins": ["@typescript-eslint"],
  "parserOptions": {
    "project": "./tsconfig.json",
    "extraFileExtensions": [".mjs"],
    "ecmaVersion": 2020
  },
  "rules": {
    "arrow-body-style": ["error", "as-needed"],
    "arrow-parens": ["error", "as-needed"],
    "brace-style": ["error", "1tbs"],
    "comma-dangle": ["error", "always-multiline"],
    "consistent-return": "off",
    "curly": ["error", "multi-line"],
    "func-names": "off",
    "import/extensions": ["error", "always"],
    "import/prefer-default-export": "off",
    "max-statements-per-line": ["error", { "max": 1 }],
    "no-console": "off",
    "no-constant-condition": ["error", { "checkLoops": false }],
    "no-continue": "off",
    "no-mixed-operators": ["error", { "allowSamePrecedence": true }],
    "no-param-reassign": ["error", { "props": false }],
    "no-plusplus": ["error", { "allowForLoopAfterthoughts": true }],
    "no-return-assign": ["error", "except-parens"],
    "no-use-before-define": ["error", { "functions": false, "classes": false }],
    "nonblock-statement-body-position": ["error", "beside"],
    "prefer-destructuring": ["error", {"object": true, "array": false}],

    "lines-around-comment": ["error", {
      "beforeBlockComment": true,
      "beforeLineComment": true,
      "allowBlockStart": true,
      "allowClassStart": true,
      "allowObjectStart": true,
      "allowArrayStart": true
    }],
    "padding-line-between-statements": [
      "error",
      { "blankLine": "always", "prev": "*", "next": "export" },
      { "blankLine": "always", "prev": "import", "next": "*" },
      { "blankLine": "any", "prev": "import", "next": "import" },
      { "blankLine": "any", "prev": "export", "next": "export" }
    ],
    "space-before-function-paren": ["error", {
      "anonymous": "never",
      "named": "never",
      "asyncArrow": "always"
    }],

    "@typescript-eslint/no-extra-parens": ["error"],
    "@typescript-eslint/no-use-before-define": ["error", { "functions": false, "classes": false }],
    "@typescript-eslint/type-annotation-spacing": ["error",
      {
        "before": false,
        "after": true,
        "overrides": {
          "arrow": {
            "before": true,
            "after": true
          }
        }
      }
    ],
    "@typescript-eslint/space-before-function-paren": ["error", {
      "anonymous": "never",
      "named": "never",
      "asyncArrow": "always"
    }]
  }
}
```

And all codes are [here](https://github.com/yubaoquan/minigrep)

**Comment by yubaoquan** [comment]
One more thing, I found the value of "main" property in eslint-config-airbnb-typescript/package.json is `dist/eslint-config-airbnb-typescript.js`, bug actually there is no such a file, and even no a dist folder.

**Comment by iamturns** [comment]
Thanks for sharing @yubaoquan 

Good catch on the `main` property, will sort that out

What command are you running to lint your project?

**Comment by yubaoquan** [comment]
I don't use lint command. I only rely on error/warning marks in vscode's editor and messages in vscode's problems panel.

**Comment by iamturns** [comment]
I've tried to replicate the issue:

- Cloned your [repo](https://github.com/yubaoquan/minigrep)
- Run `yarn install`
- Open VS Code
- View the "Problems" panel

ESLint reported no problems.

I added `const test = true` to `leetcode/ts/index.ts`, and VSCode Problems panel correctly reported the error:

<img width="682" alt="image" src="https://user-images.githubusercontent.com/719906/94922988-7bca6c80-04fe-11eb-8ae2-45a450d08c8b.png">

Any ideas on how I could replicate the issue you have?

**Comment by yubaoquan** [comment]
I checked again, the error only appears when I open files of eslint-config-airbnb-typescript it self. 

Maybe it is when I got some exception on linting my own files, then I try to find reason from lint dependencies, at this time, I opened eslint-config-airbnb-typescript's file and then vscode automatically lints it. Then it report an error. 

So I wrongly thought my linting exception is caused by this error.
